system/lib/libc/musl/src/internal/pthread_impl.h

```diff
@@ -75,39 +79,47 @@ struct __timer {
 #define _a_policy __u.__i[3*__SU+2]
 #define _a_prio __u.__i[3*__SU+3]
 #define _m_type __u.__i[0]
-#define _m_lock __u.__i[1]
-#define _m_waiters __u.__i[2]
+#define _m_lock __u.__vi[1]
+#define _m_waiters __u.__vi[2]
 #define _m_prev __u.__p[3]
 #define _m_next __u.__p[4]
 #define _m_count __u.__i[5]
 #ifdef __EMSCRIPTEN__
 #define _m_addr __u.__i[6]
 #endif
-#define _c_mutex __u.__p[0]
-#define _c_seq __u.__i[2]
-#define _c_waiters __u.__i[3]
+#define _c_shared __u.__p[0]
+#define _c_seq __u.__vi[2]
+#define _c_waiters __u.__vi[3]
 #define _c_clock __u.__i[4]
-#define _c_lock __u.__i[5]
-#define _c_lockwait __u.__i[6]
-#define _c_waiters2 __u.__i[7]
-#define _c_destroy __u.__i[8]
-#define _rw_lock __u.__i[0]
-#define _rw_waiters __u.__i[1]
+#define _c_lock __u.__vi[8]
+#define _c_head __u.__p[1]
+#define _c_tail __u.__p[5]
+#define _rw_lock __u.__vi[0]
+#define _rw_waiters __u.__vi[1]
+#define _rw_shared __u.__i[2]
+#define _b_lock __u.__vi[0]
+#define _b_waiters __u.__vi[1]
 #ifdef __EMSCRIPTEN__
 // XXX Emscripten: The spec allows detecting when multiple write locks would deadlock, so use an extra field
 // _rw_wr_owner to record which thread owns the write lock in order to avoid hangs.
 // Points to the pthread that currently has the write lock.
-#define _rw_wr_owner __u.__i[2]
+#define _rw_wr_owner __u.__vi[2]
```

Comment:
This is now incorrect. It looks like musl has added one field to the rwlock structure, old fields were `_rw_lock` at index 0, and `_rw_waiters` at index 1. In new musl there's `_rw_lock` and `_rw_waiters` at indices 0 and 1 as before, but a new field `_rw_shared` at index 2, so our Emscripten-specific deadlock detection control field at index 2 now conflicts with `_rw_shared`. We need to change `_rw_wr_owner` to index 3, and make sure that the size of that field is appropriate in Emscripten builds as well.